### PR TITLE
Made Query Types Public With Macro

### PR DESCRIPTION
**Pull request by bchong95**
# Made Query Types Public With Macro

## Description

Made Query Types Public With Macro

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Query/v3Query/QueryResponse.cs [hunk 1]
```diff
@@ -15,7 +15,14 @@ namespace Microsoft.Azure.Cosmos
     /// <summary>
     /// Represents the template class used by feed methods (enumeration operations) for the Azure Cosmos DB service.
     /// </summary>
-    internal class QueryResponse : ResponseMessage
+#if INTERNAL
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+#pragma warning disable SA1601 // Partial elements should be documented
+    public
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Query/v3Query/QueryResponse.cs]: What is the reasoning for making these public?  If compute really needs these we should be looking at using Friends instead.

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Query/v3Query/QueryResponse.cs]: You end up writing a bunch of extensions methods if that's the case.

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Query/v3Query/QueryResponse.cs]: Also needed for mocks

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Query/v3Query/QueryResponse.cs]: It makes the full type public to all of interop API's. This type is an implementation type and any contract/behavior changes impact later. 

One options is to carefully craft the required contract in friends and shadow through this implementation.

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Query/v3Query/QueryResponse.cs]: We would end up just creating a type that marshalls to every type, which is a waste of work.

File: Microsoft.Azure.Cosmos/src/FeedIteratorCore.cs [hunk 2]
```diff
@@ -17,7 +17,14 @@ namespace Microsoft.Azure.Cosmos
...
     /// </summary>
-    internal class FeedIteratorCore : FeedIteratorInternal
+#if INTERNAL
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+#pragma warning disable SA1601 // Partial elements should be documented
+    public
+#else
+    internal
+#endif
+    class FeedIteratorCore : FeedIteratorInternal
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/FeedIteratorCore.cs]: Please seal the impl classes

Comment (sboshra) [reply Microsoft.Azure.Cosmos/src/FeedIteratorCore.cs]: Please address this in another PR (to avoid re queuing this PR)

**Pull request merged**